### PR TITLE
[fix](Cloud) Use user specify storage vault first

### PR DESCRIPTION
**Pull request by ByteYue**
## Proposed changes

Issue Number: close #xxx

4.0-preview needs

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by ByteYue** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java [hunk 1]
```diff
@@ -2394,7 +2394,7 @@ private void createOlapTable(Database db, CreateTableStmt stmt) throws UserExcep
         String storageVaultName = PropertyAnalyzer.analyzeStorageVault(properties);
         String storageVaultId = null;
         // If user does not specify one storage vault then FE would use the default vault
-        if (storageVaultName == null || !storageVaultName.isEmpty()) {
+        if (storageVaultName == null || storageVaultName.isEmpty()) {
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java:2397]: Strings.isNullOrEmpty()

File: regression-test/suites/vaults/create/create.groovy [hunk 2]
```diff
@@ -61,6 +61,29 @@ suite("create_vault") {
...
+                "replication_num" = "1",
+                "storage_vault_name" = "create_hdfs_vault"
+                )
+    """
+
+    def create_stmt = """
+        show create table create_table_use_vault
+    """
+
+    assertTrue(create_stmt.contains("\"storage_vault_name\" = \"create_hdfs_vault\""))
```
Comment (gavinchou) [inline regression-test/suites/vaults/create/create.groovy:85]: also print the teh sql result `create_stmt`

**Review by pingchunzhang — APPROVED** [review]
LMGT

**Review by gavinchou — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java (hunk 1 above)
Comment (ByteYue) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java:2397]: 👍

**Comment by ByteYue** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**